Sources/ProjectSpec/Dependency.swift

```diff
@@ -36,7 +36,7 @@ public struct Dependency: Equatable {
     public enum DependencyType: Equatable {
         case target
         case framework
-        case carthage(findFrameworks: Bool?)
+        case carthage(findFrameworks: Bool?, static: Bool)
```

Comment:
1. How do you feel about making this a `linkType: LinkType` enum with `static` and `dynamic` cases? I don't feel too strongly either way. I think we @brentleyjones was building this he mentioned a similar approach.
2. Are there other dependency types that we might want use this on? If so we could move this property into the Dependency itself, as opposed to the enum

Reply:
> 1
I feel good `linkType` than the current implementation 👍 

> 2
> Are there other dependency types that we might want use this on?

No. I think it's easier to read `Dependency` has linkType property than using associated value.
But using associated value is more right to express structures.
That's difficult choices...